pylint/checkers/design_analysis.py

```diff
@@ -491,6 +513,72 @@ def leave_classdef(self, node: nodes.ClassDef) -> None:
                 args=(all_methods, self.linter.config.min_public_methods),
             )
 
+        self._check_order_dependent_resolutions(node)
+
+    def _check_order_dependent_resolutions(self, node: nodes.ClassDef) -> None:
+        if len(node.bases) < 2:
+            return
+
+        all_base_method_names: set[str] = set()
+        all_base_methods: set[nodes.FunctionDef] = set()
+        methods_defined_in_multiple_bases: set[nodes.FunctionDef] = set()
+
+        for base in node.bases:
+            klass = safe_infer(base)
+            if not isinstance(klass, nodes.ClassDef):
+                continue
+            module = get_node_first_ancestor_of_type(klass, nodes.Module)
+            assert module is not None  # TODO: remove when astroid typing improved?
+            if module.name == "typing":
+                continue
+
+            this_base_method_names = [
+                f.name for f in klass.methods() if f.name != "__init__"
+            ]
+            intersection = all_base_method_names.intersection(this_base_method_names)
+            methods_defined_in_multiple_bases.update(
+                {
+                    f
+                    for f in klass.methods()
+                    if f.name in intersection and f not in all_base_methods
+                }
+            )
+            all_base_method_names.update(this_base_method_names)
+            all_base_methods.update(klass.methods())
+
+        missing_reimplementations = [
+            m
+            for m in methods_defined_in_multiple_bases
+            if m.name not in (f.name for f in node.mymethods())
+        ]
+        if missing_reimplementations:
+            self.add_message(
+                "order-dependent-resolution",
+                node=node,
+                args=(", ".join(sorted(f.name for f in missing_reimplementations)),),
+                confidence=INFERENCE,
+            )
```

Comment:
```suggestion
        if missing_reimplementations:
            functions = ", ".join(sorted(f.name for f in missing_reimplementations))
            classes = ", ".join(b.name for b in node.bases)
            self.add_message(
                "order-dependent-resolution",
                node=node,
                args=(classes, functions),
                confidence=INFERENCE,
            )
```
(Untested but it should work ?)